### PR TITLE
PWN-4717 - recovery kit implementation

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4717

## Description of Work

PWN-4717 - RecoveryKitFragment.kt added and shown data except of phone number from api call

## Screenshot

Screenshot_20221114_155723

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/settings/model/SettingsItemMapper.kt [hunk 1]
```diff
@@ -53,10 +54,16 @@ class SettingsItemMapper(
...
         isBiometricLoginEnabled: Boolean,
-        isBiometricLoginAvailable: Boolean
+        isBiometricLoginAvailable: Boolean,
     ): List<SettingsItem> = listOfNotNull(
         SettingsGroupTitleItem(groupTitleRes = R.string.settings_item_group_title_security),
+        ComplexSettingsItem(
+            nameRes = R.string.settings_item_title_recovery_kit,
+            iconRes = R.drawable.ic_settings_shield,
+            hasSeparator = true
+        ).takeIf { isDeviceShareSaved },
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/settings/model/SettingsItemMapper.kt:66]: only if device share is saved? 
what if we restore using social+custom? i think we need to show it too

File: app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitContract.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.p2p.wallet.settings.ui.recovery
+
+import org.p2p.wallet.common.mvp.MvpPresenter
+import org.p2p.wallet.common.mvp.MvpView
+
+interface RecoveryKitContract {
+    interface View : MvpView {
+        fun showDeviceData(device: String)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitContract.kt:8]: showDeviceName(deviceName: String)
showPhoneNumber(phoneNumber: PhoneNumber / String)
showSociaId(socialId: String)

File: app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.p2p.wallet.common.mvp.BasePresenter
+
+class RecoveryKitPresenter(
+    private val signUpDetailsStorage: UserSignUpDetailsStorage,
+) : BasePresenter<RecoveryKitContract.View>(),
+    RecoveryKitContract.Presenter {
+
+    override fun attach(view: RecoveryKitContract.View) {
+        super.attach(view)
+        signUpDetailsStorage.getLastSignUpUserDetails()?.apply {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt:14]: too many `this` inside the blocks, try switching to local variables or use smth like also / let

File: app/src/main/res/layout/fragment_recovery_kit.xml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="16dp"
+        android:layout_marginTop="28dp"
+        android:text="@string/recovery_multi_factor_header"
+        android:textAppearance="@style/UiKit.TextAppearance.Regular.Caps"
+        android:textColor="@color/text_mountain"
+        app:layout_constraintTop_toBottomOf="@id/textViewRecoverySubText" />
+
+    <org.p2p.uikit.atoms.UiKitStartAmountView
+        android:id="@+id/recoveryViewDevice"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_recovery_kit.xml:87]: amountViewDevice :)

Comment (kabramovp2p) [reply app/src/main/res/layout/fragment_recovery_kit.xml:87]: naming =) i like it yes

**Review by davran1001 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/settings/model/SettingsItemMapper.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/settings/model/SettingsItemMapper.kt:66]: will fix it after API call impl

File: app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    RecoveryKitContract.Presenter {
+
+    override fun attach(view: RecoveryKitContract.View) {
+        super.attach(view)
+        signUpDetailsStorage.getLastSignUpUserDetails()?.let { userDetails ->
+            with(view) {
+                showDeviceName(
+                    Build.MANUFACTURER + " " + Build.MODEL
+                )
+                // TODO PWN-5216 show phone and device from request!
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt:19]: We don't have a request now? 
If not, just hide it. 
We are in prod, we shouldn't show stub data

File: app/src/main/res/drawable/bg_lime_rounded.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+
+    <corners android:radius="28dp" />
+
+    <solid android:color="@color/lime" />
+</shape>
```
Comment (1sakov) [inline app/src/main/res/drawable/bg_lime_rounded.xml:9]: let's name it `bg_lime_rounded_28.xml`

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt (hunk 5 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/settings/ui/recovery/RecoveryKitPresenter.kt]: fixed

File: app/src/main/res/drawable/bg_lime_rounded.xml (hunk 6 above)
Comment (kabramovp2p) [reply app/src/main/res/drawable/bg_lime_rounded.xml:9]: fixed

File: app/src/main/res/layout/fragment_recovery_kit.xml [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    android:background="@color/bg_smoke">
+
+    <org.p2p.uikit.organisms.UiKitToolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/toolbar_height"
+        android:background="@color/bg_snow"
+        app:layout_constraintTop_toTopOf="parent"
+        app:navigationIcon="@drawable/ic_toolbar_back"
+        app:navigationIconTint="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_recovery_kit.xml:17]: ic_night

**Review by 1sakov — APPROVED** [review]

**Pull request merged**